architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,241 @@
+    - Create, update, or delete a Recipe object based on the input received from the Kubernetes API Server.
+    - Create, update, or delete a Secret if it is requested by the Recipe object.
+  - **Deployment Controller**: This controller specifically watches for additions of and changes in Deployment objects in the cluster. The controller calls UCPD to perform the following operations:
+    - Create, update, or delete a Deployment object based on the input received from the Kubernetes API Server.
+    - Create, update, or delete a Secret if it is requested by the Deployment object.
+- **Recipe Validating Webhook**: This webhook is triggered by the Kubernetes API Server in case of a create, update, or delete in a Recipe object. The webhook tries to validate the action and responds to the Kubernetes API with an approval or a rejection. For more information about webhooks, refer to the [official documentation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+- **Health Checks**: Health checks are implemented to monitor the status and performance of the controllers. They ensure that the controllers are functioning correctly and can take corrective actions if any issues are detected.
+
+### Clients
+
+In this section, we will discuss the different clients of the Controller component. Clients are systems that interact with the Controller component to trigger actions. Here are the clients of the Controller component:
+
+1. **Kubernetes API Server**: The primary client that interacts with the controller. It communicates with the validating webhook and the controllers in case of resource changes (e.g., creation, update, or deletion of a Recipe or Deployment) requested by another interactor (for example; a human running `kubectl` commands). The controllers watch for these calls and reconcile the state of the resources accordingly.
+1. **Health Check Probes**: Kubernetes itself can act as a client by performing health and readiness checks on the controller manager.
+1. **Metrics Scrapers**: If metrics are enabled, Prometheus or other monitoring tools can scrape metrics from the controller manager.
+
+## Trust Boundaries
+
+We have a few different trust boundaries for the Controller component:
+
+- **Kubernetes Cluster**: The overall environment where the Controller component operates and receives requests from the clients.
+- **Namespaces within the Cluster**: Logical partitions within the cluster to separate and isolate resources and workloads.
+
+The Controller component lives inside the `radius-system` namespace in the Kubernetes cluster where it is installed. UCPD also resides within the same namespace.
+
+The Kubernetes API Server, which is the main interactor of the Controller component, runs in the `kube-system` namespace within the cluster.
+
```

Comment:
What are the security implications of namespaces? Can you elaborate on why this is relevant information in the context of security?

Reply:
Added below in the **Key Points of Namespaces** section.